src/ImageSharp/Formats/Png/PngEncoderCore.cs

```diff
@@ -238,22 +260,17 @@ public void Encode<TPixel>(Image<TPixel> image, Stream stream, CancellationToken
                 }
 
                 // Each frame control sequence number must be incremented by the number of frame data chunks that follow.
-                frameControl = this.WriteFrameControlChunk(stream, frameMetadata, bounds, (uint)i + increment);
```

Comment:
There's a TODO in `WriteFrameControlChunk` where we need to set the X/Y offset. I'm not sure how your tests can pass without it being implemented.

Reply:
I'm not seeing this TODO? Neither of the two TODOs in PngEncoderCore have anything to do with that, as far as I can tell. Where is this? WriteFrameControlChunk does write x/y offsets.

I didn't have a test for it, but I just added one locally and it made a visually correct image, and looking at the metadata in TweakPNG it did use frame offsets. Not exactly the same ones as the original, but that's just an encoding difference.
It also set off ImageComparer.Exact, with the largest frame difference at 0.0024%. I think this is inherent to the current process of decoding and encoding, since compositing happens during decoding.

Reply:
Ah yes, it's fixed in the main branch but will need to be fixed in the v3 branch.

Reply:
Looks like it was already fixed in release/3.1.x, by commit 7890008

Reply:
Yeah. Sorry, I was looking at a separate old branch.